### DISCUSSION TITLE
Controlling KeyboardControls Programatically

### DISCUSSION THREAD

**Discussion by timhc22** [pmndrs/drei#1267] [General]
Is it possible to use a 'set' method on KeyboardControls so as to simulate a keyboard press:

```
const [, get] = useKeyboardControls()  
const { forward, backward, left, right, jump, action } = get()
```

I'm currently using as above, and can detect when a keyboard button is pressed, but I want to create 'buttons' for mobile which will simulate the keyboard presses, but am not quite sure how to do it.

There seems to be a `sub` function but I'm not sure what it really does or how to use it:
https://github.com/pmndrs/drei/blob/master/src/web/KeyboardControls.tsx

**Comment by Seanmclem** [comment]
@timhc22 did you ever figure this out? Just asked my own in Q&A
